src/deepqmc/wf/paulinet/schnet.py

```diff
@@ -200,26 +211,28 @@ class ElectronicSchNet(nn.Module):
     layer_factories = {
         1: SchNetLayer,
         2: SchNetSpinLayer,
+        'mean-field': SchNetSingleElectronLayer,
     }
 
     def __init__(
         self,
         n_up,
         n_down,
         n_nuclei,
-        embedding_dim,
         dist_feat_dim,
         subnet_metafactory=None,
         *,
         n_interactions=3,
         kernel_dim=64,
+        embedding_dim=64,
         version=2,
         layer_norm=False,
     ):
         assert version in self.layer_factories
         subnet_metafactory = subnet_metafactory or SubnetFactory
         subnet_factory = subnet_metafactory(dist_feat_dim, kernel_dim, embedding_dim)
         super().__init__()
+        self.embedding_dim = embedding_dim
```

Comment:
This shouldn't be here, since it's not used by the actual object. If some upstream object needs it, the embedding dim should be passed down from that upstream object

Reply:
That is true, however when a different network is used to generate the embeddings it might not have the same submodules, but it certainly has an embedding dimensions. That is to say this should make the architecture more modular.